ui/src/app/shared/utils/url-search-params.utils.ts

```diff
@@ -95,11 +124,17 @@ export class URLSearchParamsUtils {
   }
 
   /** Returns the list of queryable non-label fields. */
-  public static getQueryFields(): string[] {
-    let fields: string[] = [];
-    for (let field in queryFields) {
-      fields.push(queryFields[field]);
+  public static getQueryFields(capabilities: CapabilitiesResponse): Map<string, FieldDataType>  {
+    // We could make commonLabels a map of label to FieldDataType to support
+    // non-text label types.
+    var labelFields = new Map<string, FieldDataType>();
+    if (capabilities.commonLabels) {
+      labelFields = new Map<string, FieldDataType>(
+        capabilities.commonLabels.map(l => [l, FieldDataType.Text] as [string, FieldDataType]));
     }
-    return fields;
+    return new Map<string, FieldDataType>([
+      ...Array.from(queryDataTypes),
```

Comment:
What is this `...` doing?

Reply:
`...` is the spread operator https://github.com/tc39/proposal-object-rest-spread 

Basically its just merging all three maps into a new map.

Reply:
Oh that’s interesting! Thanks for clarifying!